CHANGELOG.md

```diff
@@ -24,6 +24,10 @@ Current
 
 ### Changed:
 
+- [`DimensionStoreKeyUtils` now supports case sensitive row and column keys](https://github.com/yahoo/fili/pull/90)
```

Comment:
Could we add a blurb to the `### Deprecated` section to, calling out that case-insensitive row retrieval is deprecated? It's unlikely that anyone really cares, but having it in the deprecated section will call a bit more attention to the change as well